modules/odo-core-concepts.adoc

```diff
@@ -0,0 +1,46 @@
+// Module included in the following assemblies:
+//
+// * cli_reference/developer_cli_odo/understanding-odo.adoc
+
+:_content-type: CONCEPT
+[id="odo-core-concepts_{context}"]
+
+= odo core concepts
+
+`{odo-title}` abstracts Kubernetes concepts into a developer friendly terminology:
+
+Application::
+An _application_ in `odo` is a classic application, developed with a link:https://www.redhat.com/en/topics/cloud-native-apps[cloud-native approach], that is used to perform a particular task.
++
+Examples of applications: Online Video Streaming, Hotel Reservation System, Online Shopping.
+
+Component::
+A cloud-native application is a collection of small, independent, loosely coupled _components_. An `odo` component is a set of Kubernetes resources that  can run and be deployed separately.
++
+Examples of components: API Backend, Web Frontend, Payment Backend.
+
+Project::
+A _project_ contains your source code, tests, and libraries, organized in a single unit. Projects helps achieve multi-tenancy, allowing multiple teams to run multiple applications at the same time, on a single cluster.
+
+
+Context::
+The _context_ is a directory that contains the source code, tests, libraries and `odo` specific config files for a single component.
+
+URL::
+A _URL_ exposes a component for access from outside the cluster.
+
+Storage::
+_Storage_ is the persistent storage in the cluster. It persists the data across restarts and across any rebuilds of a component.
+
+Service::
+A _service_ is an external application that provides additional functionality to a component.
++
+Examples of services:  PostgreSQL, MySQL, Redis, RabbitMQ.
++
+In `{odo-title}`, services are provisioned from the OpenShift Service Catalog and must be enabled within your cluster.
+
+
+devfile::
+The _devfile_ is an open standard for defining containerized development environments that enables developer tools to simplify and accelerate workflows. For more information, see the documentation at link:https://devfile.io[].
++
+You can connect to publicly available devfile registries, or you can install your own Secure Registry. 
```

Comment:
```suggestion
You can connect to publicly available devfile registries or install a Secure Registry. 
```

Reply:
The reason I add in the extra "or you can" is to make it easier to parse - otherwise, you read "you can connect to publicly" ... "or install...", and it can be hard to know where to split the first phrase. So in this case, the extra words help, IMO.
 
Agreed on removing "your own"  Secure Registry